### DISCUSSION TITLE
Problems with creating database on Clickhouse

### DISCUSSION THREAD

**Discussion by Jeroenvb3** [plausible/analytics#2319] [Self-Hosted Support]
I am trying to install Plausible according to the Self-Hosted docs: [docs](https://plausible.io/docs/self-hosting).

When I start the containers with `docker-compose up -d` it doesn't work. The logs show the following:
```
plausible_1            | ** (MatchError) no match of right hand side value: {:error, "connection not available and request was dropped from queue after 2000ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:\n\n  1. By tracking down slow queries and making sure they are running fast enough\n  2. Increasing the pool_size (albeit it increases resource consumption)\n  3. Allow requests to wait longer by increasing :queue_target and :queue_interval\n\nSee DBConnection.start_link/2 for more information\n"}
plausible_1            |     (plausible 0.0.1) lib/plausible_release.ex:48: anonymous fn/2 in Plausible.Release.createdb/0
plausible_1            |     (elixir 1.12.2) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
plausible_1            |     (plausible 0.0.1) lib/plausible_release.ex:47: Plausible.Release.createdb/0
plausible_1            |     (stdlib 3.15) erl_eval.erl:685: :erl_eval.do_apply/6
plausible_1            |     (elixir 1.12.2) lib/code.ex:656: Code.eval_string_with_error_handling/3

```

suggesting Clickhouse can't handle creating a database, which seems weird to me.

What could be causing this, or how do I get around this problem?

Info:

I am running it on a local KVM running Ubuntu 22.04 with kernel 5.15.0-50-generic.

**Comment by ruslandoga** [comment]
@Jeroenvb3 👋

It is weird indeed. Do you still have access to full `docker compose` logs? Could you please post them here?

**Reply by Jeroenvb3** [reply]
@ruslandoga thanks for you interest. 

If I tail the logs, it shows the message I've already send over and over (it keeps restarting the docker).

I have now started a new VM (Ubuntu 22.04 Desktop (instead of server)). And I have installed `docker compose` instead of `docker-compose` (`sudo apt install docker-compose-plugin`). I'm not entirely sure how much difference it makes but it works now.

**Reply by ruslandoga** [reply]
I was more interested in the logs from the other containers :)
But it doesn't matter now, I guess.

**Comment by maesitos** [comment]
In my case `aa-remove-unknown` solved the problem.

More information here:
https://javahowtos.com/guides/124-docker/414-solved-cannot-kill-docker-container-permission-denied.html

**Comment by rbndg** [comment]
I am having the same issue. the error above just loops over and over again

**Reply by ruslandoga** [reply]
👋 @rbndg 

Could you please describe your setup in details? Where is ClickHouse hosted, how Plausible is deployed, etc.

**Reply by rbndg** [reply]
Ubuntu 20.04.1 LTS
Docker version 24.0.2, build cb74dfc

**Reply by rbndg** [reply]
I am just following:
https://plausible.io/docs/self-hosting

**Reply by rbndg** [reply]
I was able to fix it by reinstalling docker from scratch. I dont know why this was happening

**Reply by maesitos** [reply]
> I was able to fix it by reinstalling docker from scratch. I dont know why this was happening

I did manage to solve it by uninstalling AppArmor, that also reinstalls Docker. I suggest backing up the volumes first with https://github.com/BretFisher/docker-vackup

**Comment by ruslandoga** [comment]
Might be related: https://github.com/plausible/analytics/discussions/2875
